src/main/java/com/minecolonies/coremod/entity/ai/mobs/util/MobEventsUtils.java

```diff
@@ -107,6 +122,63 @@ else if (hordeTotal > MAX_SIZE)
         }
     }
 
+    private static BlockPos calculateFallBackSpawnLocation(final World world, final Colony colony)
+    {
+        if (colony == null)
+        {
+            Log.getLogger().error("When trying to find SpawnPoint for Barbarian Event, the Colony was Null!");
+            return null;
+        }
+        final BlockPos center = colony.getCenter();
+        final int radius = Configurations.workingRangeTownHall;
+
+        final int random = world.rand.nextInt(4);
+
+        final int x;
+        final int y;
+        final int z;
+
+        if (random == 0)
+        {
+            x = center.getX() + radius;
+            y = center.getY();
+            z = center.getZ();
+        }
+        else if (random == 1)
+        {
+            x = center.getX() - radius;
+            y = center.getY();
+            z = center.getZ();
+        }
+        else if (random == 2)
+        {
+            x = center.getX();
+            y = center.getY();
+            z = center.getZ() + radius;
+        }
+        else if (random == 3)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)